### PR TITLE
:tada: Source Pinterest - remove odd auth method

### LINKED ISSUE #19235: Source-Pinterest: Check method returns error while using access token authentication in config
## Environment
- **OS Version / Instance**: macOS
- **Deployment**: Docker 
- **Source Connector and version**: source-pinterest:0.1.9
- **Step where error happened**: Setup new connection 

## Current Behavior
Returns failure status with an error message

## Expected Behavior
Returns `{"type": "CONNECTION_STATUS", "connectionStatus": {"status": "SUCCEEDED"}}`

## Logs
```
{"type": "LOG", "log": {"level": "ERROR", "message": "Check failed"}}
{"type": "CONNECTION_STATUS", "connectionStatus": {"status": "FAILED", "message": "TypeError(\"unsupported operand type(s) for +: 'NoneType' and 'str'\")"}}

```
## Steps to Reproduce
1. Create config file for pinterest with access_token authentication type
2. Run the check method

### PR DESCRIPTION
**Pull request by midavadim**
## What
https://github.com/airbytehq/airbyte/issues/19235
https://github.com/airbytehq/oncall/issues/3001
https://github.com/airbytehq/oncall/issues/2258

error "unsupported operand type(s) for +: 'NoneType' and 'str'" happens when 'Token' authentication is choosen.

## How
Research showed that our code never supported "access_token" based authentication.
I've tried to implement it but "access_token" provided via pinterest UI is:
- a short-term token (expires in 1 day) and 
- is limited to 3 streams only (has limited scope).

So I decided that this method should be removed at all.

## 🚨 User Impact 🚨
Even though we removed the authentication method, there is no user who could use it becuase it never worked.

## Pre-merge Actions
*Expand the relevant checklist and delete the others.*

<details><summary><strong>New Connector</strong></summary>

### Community member or Airbyter

- **Community member?** Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- Unit & integration tests added and passing. Community members, please provide proof of success locally e.g: screenshot or copy-paste unit, integration, and acceptance test output. To run acceptance tests for a Python connector, follow instructions in the README. For java connectors run `./gradlew :airbyte-integrations:connectors:<name>:integrationTest`.
- Connector version is set to `0.0.1`
    - `Dockerfile` has version `0.0.1`
- Documentation updated
    - Connector's `README.md`
    - Connector's `bootstrap.md`. See [description and examples](https://docs.google.com/document/d/1ypdgmwmEHWv-TrO4_YOQ7pAJGVrMp5BOkEVh831N260/edit?usp=sharing)
    - `docs/integrations/<source or destination>/<name>.md` including changelog with an entry for the initial version. See changelog [example](https://docs.airbyte.io/integrations/sources/stripe#changelog)
    - `docs/integrations/README.md`

### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items.

- Create a non-forked branch based on this PR and test the below items on it
- Build is successful
- If new credentials are required for use in CI, add them to GSM. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci).

</details>

<details><summary><strong>Updating a connector</strong></summary>

### Community member or Airbyter

- Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- Unit & integration tests added

### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items.

- Create a non-forked branch based on this PR and test the below items on it
- Build is successful
- If new credentials are required for use in CI, add them to GSM. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci).

</details>

<details><summary><strong>Connector Generator</strong></summary>

- Issue acceptance criteria met
- PR name follows [PR naming conventions](https://docs.airbyte.com/contributing-to-airbyte/resources/pull-requests-handbook)
- If adding a new generator, add it to the [list of scaffold modules being tested](https://github.com/airbytehq/airbyte/blob/master/airbyte-integrations/connector-templates/generator/build.gradle#L41)
- The generator test modules (all connectors with `-scaffold` in their name) have been updated with the latest scaffold by running `./gradlew :airbyte-integrations:connector-templates:generator:generateScaffolds` then checking in your changes
- Documentation which references the generator is updated as needed

</details>

<details><summary><strong>Updating the Python CDK</strong></summary>

### Airbyter

Before merging:
- Pull Request description explains what problem it is solving
- Code change is unit tested
- Build and my-py check pass
- Smoke test the change on at least one affected connector
   - On Github: Run [this workflow](https://github.com/airbytehq/airbyte/actions/workflows/connectors_tests.yml), passing `--use-local-cdk --name=source-<connector>` as options
   - Locally: `airbyte-ci connectors --use-local-cdk --name=source-<connector> test`
- PR is reviewed and approved
      
After merging:
- [Publish the CDK](https://github.com/airbytehq/airbyte/actions/workflows/publish-cdk-command-manually.yml)
   - The CDK does not follow proper semantic versioning. Choose minor if this the change has significant user impact or is a breaking change. Choose patch otherwise.
   - Write a thoughtful changelog message so we know what was updated.
- Merge the platform PR that was auto-created for updating the Connector Builder's CDK version
   - This step is optional if the change does not affect the connector builder or declarative connectors.

</details>

### CODE REVIEW

**Comment by airbyte-oss-build-runner** [comment]
##  source-pinterest test report (commit [`2f00feb7c0`](https://github.com/airbytehq/airbyte/pull/31876/commits/2f00feb7c08d7bcb07e140e122a4fb89d775c089)) - ❌

⏲️  Total pipeline duration: 31.90s 

| Step                                                            | Result   |
|:----------------------------------------------------------------|:---------|
| Build source-pinterest docker image for platform(s) linux/amd64 | ✅       |
| Unit tests                                                      | ❌       |
| Code format checks                                              | ✅       |
| Validate metadata for source-pinterest                          | ❌       |
| Connector version semver check                                  | ✅       |
| Connector version increment check                               | ✅       |
| QA checks                                                       | ❌       |

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/midavadim_31705-pinterest-remove-odd-auth-method/1698335088/2f00feb7c08d7bcb07e140e122a4fb89d775c089/source-pinterest/1.0.0/output.html)

☁️ [View runs for commit in Dagger Cloud](https://alpha.dagger.cloud/changeByPipelines?filter=dagger.io/git.ref:2f00feb7c08d7bcb07e140e122a4fb89d775c089)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/airbyte-ci/connectors/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=source-pinterest test
```

**Comment by airbyte-oss-build-runner** [comment]
##  source-pinterest test report (commit [`5edfeca4b3`](https://github.com/airbytehq/airbyte/pull/31876/commits/5edfeca4b39981652fcccaa74d4423133596f1ca)) - ❌

⏲️  Total pipeline duration: 42.21s 

| Step                                                            | Result   |
|:----------------------------------------------------------------|:---------|
| Build source-pinterest docker image for platform(s) linux/amd64 | ✅       |
| Unit tests                                                      | ❌       |
| Code format checks                                              | ✅       |
| Validate metadata for source-pinterest                          | ✅       |
| Connector version semver check                                  | ✅       |
| Connector version increment check                               | ✅       |
| QA checks                                                       | ✅       |

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/midavadim_31705-pinterest-remove-odd-auth-method/1698336794/5edfeca4b39981652fcccaa74d4423133596f1ca/source-pinterest/0.7.0/output.html)

☁️ [View runs for commit in Dagger Cloud](https://alpha.dagger.cloud/changeByPipelines?filter=dagger.io/git.ref:5edfeca4b39981652fcccaa74d4423133596f1ca)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/airbyte-ci/connectors/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=source-pinterest test
```

**Review by bnchrch — CHANGES_REQUESTED** [review]
Just need a description! Im curious if removing this auth will impact customers?

**Comment by airbyte-oss-build-runner** [comment]
##  source-pinterest test report (commit [`8c8ec04622`](https://github.com/airbytehq/airbyte/pull/31876/commits/8c8ec046222f82f7a3aa33391f2117b0c98dce7b)) - ❌

⏲️  Total pipeline duration: 03mn12s 

| Step                                                            | Result   |
|:----------------------------------------------------------------|:---------|
| Build source-pinterest docker image for platform(s) linux/amd64 | ✅       |
| Unit tests                                                      | ✅       |
| Acceptance tests                                                | ❌       |
| Check our base image is used                                    | ✅       |
| Code format checks                                              | ❌       |
| Validate metadata for source-pinterest                          | ✅       |
| Connector version semver check                                  | ✅       |
| Connector version increment check                               | ✅       |
| QA checks                                                       | ✅       |

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/midavadim_31705-pinterest-remove-odd-auth-method/1698337455/8c8ec046222f82f7a3aa33391f2117b0c98dce7b/source-pinterest/0.7.0/output.html)

☁️ [View runs for commit in Dagger Cloud](https://alpha.dagger.cloud/changeByPipelines?filter=dagger.io/git.ref:8c8ec046222f82f7a3aa33391f2117b0c98dce7b)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/airbyte-ci/connectors/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=source-pinterest test
```

**Comment by airbyte-oss-build-runner** [comment]
##  source-pinterest test report (commit [`e7b4978e1c`](https://github.com/airbytehq/airbyte/pull/31876/commits/e7b4978e1cd19273094c0de34a53338a8576c312)) - ❌

⏲️  Total pipeline duration: 01mn13s 

| Step                                                            | Result   |
|:----------------------------------------------------------------|:---------|
| Build source-pinterest docker image for platform(s) linux/amd64 | ✅       |
| Unit tests                                                      | ✅       |
| Acceptance tests                                                | ❌       |
| Check our base image is used                                    | ✅       |
| Code format checks                                              | ✅       |
| Validate metadata for source-pinterest                          | ✅       |
| Connector version semver check                                  | ✅       |
| Connector version increment check                               | ✅       |
| QA checks                                                       | ✅       |

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/midavadim_31705-pinterest-remove-odd-auth-method/1698338353/e7b4978e1cd19273094c0de34a53338a8576c312/source-pinterest/0.7.0/output.html)

☁️ [View runs for commit in Dagger Cloud](https://alpha.dagger.cloud/changeByPipelines?filter=dagger.io/git.ref:e7b4978e1cd19273094c0de34a53338a8576c312)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/airbyte-ci/connectors/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=source-pinterest test
```

File: airbyte-integrations/connectors/source-pinterest/source_pinterest/schemas/ad_account_analytics.json [hunk 1]
```diff
@@ -1,5 +1,5 @@
 {
-  "$schema": "http://json-schema.org/draft-07/schema#",
+  "$schema": "httpss://json-schema.org/draft-07/schema#",
```
Comment (davydov-d) [inline airbyte-integrations/connectors/source-pinterest/source_pinterest/schemas/ad_account_analytics.json:2]: typo

File: airbyte-integrations/connectors/source-pinterest/source_pinterest/schemas/ad_accounts.json [hunk 2]
```diff
@@ -1,5 +1,5 @@
 {
-  "$schema": "http://json-schema.org/draft-07/schema#",
+  "$schema": "httpss://json-schema.org/draft-07/schema#",
```
Comment (davydov-d) [inline airbyte-integrations/connectors/source-pinterest/source_pinterest/schemas/ad_accounts.json:2]: same here

File: airbyte-integrations/connectors/source-pinterest/source_pinterest/schemas/ad_analytics.json [hunk 3]
```diff
@@ -1,5 +1,5 @@
 {
-  "$schema": "http://json-schema.org/draft-07/schema#",
+  "$schema": "httpss://json-schema.org/draft-07/schema#",
```
Comment (davydov-d) [inline airbyte-integrations/connectors/source-pinterest/source_pinterest/schemas/ad_analytics.json:2]: one more

File: airbyte-integrations/connectors/source-pinterest/source_pinterest/schemas/ad_groups.json [hunk 4]
```diff
@@ -1,5 +1,5 @@
 {
-  "$schema": "http://json-schema.org/draft-07/schema#",
+  "$schema": "httpss://json-schema.org/draft-07/schema#",
```
Comment (davydov-d) [inline airbyte-integrations/connectors/source-pinterest/source_pinterest/schemas/ad_groups.json:2]: and more

Comment (davydov-d) [inline airbyte-integrations/connectors/source-pinterest/source_pinterest/schemas/ad_groups.json:2]: and more

**Review by davydov-d — CHANGES_REQUESTED** [review]
please fix typos and acceptance tests

File: airbyte-integrations/connectors/source-pinterest/source_pinterest/schemas/ad_accounts.json (hunk 2 above)
Comment (midavadim) [reply airbyte-integrations/connectors/source-pinterest/source_pinterest/schemas/ad_accounts.json]: fixed

File: airbyte-integrations/connectors/source-pinterest/source_pinterest/schemas/ad_groups.json (hunk 4 above)
Comment (midavadim) [reply airbyte-integrations/connectors/source-pinterest/source_pinterest/schemas/ad_groups.json]: fixed

**Comment by midavadim** [comment]
> Just need a description! Im curious if removing this auth will impact customers?
@bnchrch 
Sorry, I was struggling with tests and has not updated PR properly at that time.

The description has been added now

**Review by davydov-d — APPROVED** [review]

**Review by bnchrch — APPROVED** [review]
Amazing stuff and thank you for updating to use the new base image

**Pull request merged**